extension/js/common/core/crypto/pgp/openpgp-key.ts

```diff
@@ -269,114 +244,99 @@ export class OpenPGPKey {
+    if (flags & opgp.enums.keyFlags.encryptStorage) {
       strs.push('encrypt_storage');
     }
-    if (flags & opgp.enums.keyFlags.sign_data) {
+    if (flags & opgp.enums.keyFlags.signData) {
       strs.push('sign_data');
     }
-    if (flags & opgp.enums.keyFlags.certify_keys) {
+    if (flags & opgp.enums.keyFlags.certifyKeys) {
       strs.push('certify_keys');
     }
     return '[' + strs.join(', ') + ']';
   };
 
   public static diagnose = async (pubkey: Key, passphrase: string): Promise<Map<string, string>> => {
-    const key = OpenPGPKey.extractExternalLibraryObjFromKey(pubkey);
+    const key = await OpenPGPKey.extractExternalLibraryObjFromKey(pubkey);
     const result = new Map<string, string>();
-    if (!key.isPrivate() && !key.isPublic()) {
-      result.set(`key is neither public or private!!`, '');
-      return result;
-    }
     result.set(`Is Private?`, KeyUtil.formatResult(key.isPrivate()));
     for (let i = 0; i < key.users.length; i++) {
       // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      result.set(`User id ${i}`, key.users[i].userId!.userid);
+      result.set(`User id ${i}`, key.users[i].userID!.userID);
     }
     const user = await key.getPrimaryUser();
-    result.set(`Primary User`, user?.user?.userId?.userid || 'No primary user');
+    result.set(`Primary User`, user?.user?.userID?.userID || 'No primary user');
     result.set(`Fingerprint`, Str.spaced(key.getFingerprint().toUpperCase() || 'err'));
     // take subkeys from original key so we show subkeys disabled by #2715 too
-    const subKeys = OpenPGPKey.extractStrengthUncheckedExternalLibraryObjFromKey(pubkey)?.subKeys ?? key.subKeys;
+    const subKeys = (await OpenPGPKey.extractStrengthUncheckedExternalLibraryObjFromKey(pubkey))?.subkeys ?? key.subkeys;
     result.set(`Subkeys`, KeyUtil.formatResult(subKeys ? subKeys.length : subKeys));
-    result.set(`Primary key algo`, KeyUtil.formatResult(key.primaryKey.algorithm));
+    result.set(`Primary key algo`, KeyUtil.formatResult(opgp.enums.read(opgp.enums.publicKey, key.keyPacket.algorithm)));
     const flags = await OpenPGPKey.getPrimaryKeyFlags(key);
     result.set(`Usage flags`, KeyUtil.formatResult(OpenPGPKey.keyFlagsToString(flags)));
-    if (key.isPrivate() && !key.isFullyDecrypted()) {
+    if (key.isPrivate() && !OpenPGPKey.isFullyDecrypted(key)) {
       result.set(
         `key decrypt`,
         await KeyUtil.formatResultAsync(async () => {
-          try {
-            await key.decrypt(passphrase); // throws on password mismatch
-            return 'success';
-          } catch (e) {
-            if (e instanceof Error && e.message.toLowerCase().includes('incorrect key passphrase')) {
-              return 'INCORRECT PASSPHRASE';
-            } else {
-              throw e;
-            }
-          }
+          return (await OpenPGPKey.decryptPrivateKey(key, passphrase)) ? 'success' : 'INCORRECT PASSPHRASE';
```

Comment:
Are other error types ignored intentionally here? As previously errors were rethrown

Reply:
If an exception occurs, `formatResultAsync` will display the exception data in square brackets
```
    } catch (e) {
      return `[${String(e)}]`;
    }
```